docs/guide/best-practices.md

```diff
@@ -1,40 +1,35 @@
+Using `target` to manage multiple environments gives you the flexibility set up your environments how you choose. Commonly, environments are managed by schemas within the same database: all test models are deployed to a schema called `dbt_[username]` and production models are deployed to a schema called `analytics`. An ideal setup would have production and test databases completely separate. Either way, we highly recommend maintaining multiple environments and managing deployments with `target`.
 
 ## Source control workflows
 
-We believe that all dbt projects should be managed via source control. We use git for all of our source control, and use branching and pull requests to keep the master branch the sole source of organizational truth. Future versions of dbt will include hooks that will automatically deploy to production upon pushing to master.
+We believe that all dbt projects should be managed via source control. We use git for all of our source control, and use branching and pull requests to keep the master branch the sole source of organizational truth.
 
 ## Using dbt interactively
 
-The best development tools allow for very small units of work to be developed and tested quickly. One of the major advantages of dbt is getting analytics out of clunky tools and into text files that can be edited in whatever your editor of choice is—we have folks using vim, emacs, and Atom.
-
-When your project gets large enough, `dbt run` can begin to take a while. This stage in your development could be a bottleneck and slow you down. dbt provides three primary ways to address this:
+When your project gets large enough, `dbt run` can take a while. dbt provides three primary ways to address this so that you can deploy changes to your database quickly:
 
 1. Use views instead of tables to the greatest extent possible in development. Views typically deploy much faster than tables, and in development it's often not critical that subsequent analytic queries run as fast as possible. It's easy to change this setting later and it will have no impact on your business logic.
 1. Use `dbt_project.yml` to disable portions of your project that you're not currently working on. If you have multiple modules within a given project, turn off the ones that you're not currently working on so that those models don't deploy with every `dbt run`.
-1. Pass the `--model` flag to `dbt run`. This flag asks dbt to only `run` the models you specify and their dependents. If you're working on a particular model, this can make a very significant difference in your workflow.
+1. Pass the `--models` flag to `dbt run`. This flag causes dbt to only deploy the models you specify and their dependents. If you're working on a particular model, this can make a very significant difference in your deployment time in large projects.
+1. [mention limitation based on `target` once that is built]
```

Comment:
👍 

we have an issue for this, added a note to it:
https://github.com/analyst-collective/dbt/issues/149